### ISSUE TITLE
Use of regex in SPARQL

### ISSUE THREAD

**Issue by davidshumway** [openlink/virtuoso-opensource#1015]
Is use of regex pattern matching in SPARQL enabled?

I see there's some regex documentation but it appears to be for SQL only. Is this correct? e.g. (http://docs.openlinksw.com/virtuoso/fn_regexp_match/)

**Comment by Tomas2D** [comment]
The syntax is `FILTER regex(?x, "pattern" [, "flags"])`

so for example 
```sparql
PREFIX geo: <http://www.opengis.net/ont/geosparql#>
PREFIX osmt: <https://www.openstreetmap.org/wiki/Key:>

SELECT ?roadName ?roadGeo
WHERE {
  ?roadIRI osmt:highway ?roadType ;
        osmt:name ?roadName ;
        geo:hasGeometry ?roadGeo .
  
  FILTER (REGEX(?roadGeo, "^LINESTRING" ))
}
```

**Comment by TallTed** [comment]
Yes, regex in SPARQL is supported, based on the [SPARQL REGEX syntax](https://www.w3.org/TR/2013/REC-sparql11-query-20130321/#func-regex) (as @Tomas2D provided), and using [Perl Compatible Regular Expressions](https://en.wikipedia.org/wiki/Perl_Compatible_Regular_Expressions).  You may find some of [the discussion here](https://stackoverflow.com/questions/40437602/what-flavor-of-regex-should-be-used-with-dbpedia-virtuoso-sparql) to be helpful, depending on what you're trying to match.
